controllers/keystoneservice_controller.go

```diff
@@ -204,7 +204,23 @@ func (r *KeystoneServiceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 	reconcileEndpoint(identityClient, serviceID, instance.Spec.ServiceName, instance.Spec.Region, "internal", instance.Spec.InternalURL)
 	reconcileEndpoint(identityClient, serviceID, instance.Spec.ServiceName, instance.Spec.Region, "public", instance.Spec.PublicURL)
 
-	err = reconcileUser(reqLogger, identityClient, instance.Spec.ServiceName)
+	var username string
+	if instance.Spec.Username == "" {
+		username = instance.Spec.ServiceName
+	} else {
+		username = instance.Spec.Username
+	}
+
+	var password string
+	if instance.Spec.Password == "" {
```

Comment:
shouldn't this be:
`if instance.Spec.Password != ""`

Reply:
yea, fixed that